File: src/test/java/cn/enaium/cf4m/test/T.java
package cn.enaium.cf4m.test;

import cn.enaium.cf4m.CF4M;
import cn.enaium.cf4m.test.config.ConfigTest;
import cn.enaium.cf4m.test.event.EventCancelTest;
import cn.enaium.cf4m.test.event.EventPriorityTest;
import cn.enaium.cf4m.test.event.EventRegisterUnregisterTest;
import cn.enaium.cf4m.test.module.ModuleBeanTest;
import org.junit.jupiter.api.Test;

/**
 * Project: cf4m
 * Author: Enaium
 */
public class T {
    @Test
    public void test() {
        CF4M.run(T.class, System.getProperty("user.dir") + "/build/configTest");
        System.out.println("Check ModuleBeanTest Start");
        new ModuleBeanTest();
        System.out.println("Check ModuleBeanTest End");
        System.out.println();
        System.out.println("Check EventRegisterUnregister Start");
        new EventRegisterUnregisterTest();
        System.out.println("Check EventRegisterUnregister End");
        System.out.println();
        System.out.println("Check EventPriorityTest Start");
        new EventPriorityTest();
        System.out.println("Check EventPriorityTest End");
        System.out.println();
        System.out.println("Check EventCancelTest Start");
        new EventCancelTest();
        System.out.println("Check EventCancelTest End");
        System.out.println();
        System.out.println("Check ModuleEnableDisableTest Start");
        CF4M.getModule().onKey(1);
        CF4M.getModule().onKey(1);
        System.out.println("Check ModuleEnableDisableTest End");
        System.out.println();
        System.out.println("Check ModuleExtendTest Start");
        CF4M.getModule().onKey(2);
        CF4M.getModule().onKey(2);
        System.out.println("Check ModuleExtendTest1 End");
        System.out.println();
        CF4M.getModule().onKey(3);
        CF4M.getModule().onKey(3);
        System.out.println("Check ModuleExtendTest1 End");
        System.out.println();
        System.out.println("Check ModuleSettingTest Start");
        CF4M.getModule().onKey(4);
        System.out.println("Check ModuleSettingTest End");
        System.out.println();
        System.out.println("Check CommandTest Start");
        System.out.println(CF4M.getCommand().execCommand("`t"));
        System.out.println(CF4M.getCommand().execCommand("`t var1"));
        System.out.println(CF4M.getCommand().execCommand("`t var1 1"));
        System.out.println("Check CommandTest End");
        System.out.println();
        System.out.println("Check ConfigTest Start");
        new ConfigTest();
        System.out.println("Check ConfigTest End");
    }
}

File: src/test/java/cn/enaium/cf4m/test/T.kt
package cn.enaium.cf4m.test

import org.junit.jupiter.api.Test
import cn.enaium.cf4m.CF4M.run
import cn.enaium.cf4m.CF4M.INSTANCE
import cn.enaium.cf4m.test.module.ModuleBeanTest
import cn.enaium.cf4m.test.event.EventRegisterUnregisterTest
import cn.enaium.cf4m.test.event.EventPriorityTest
import cn.enaium.cf4m.test.event.EventCancelTest
import cn.enaium.cf4m.test.config.ConfigTest

/**
 * Project: cf4m
 *
 * @author Enaium
 */
class T {
    @Test
    fun test() {
        run(this, System.getProperty("user.dir") + "/build/configTest")
        println("Check ModuleBeanTest Start")
        ModuleBeanTest()
        println("Check ModuleBeanTest End")
        println()
        println("Check EventRegisterUnregister Start")
        EventRegisterUnregisterTest()
        println("Check EventRegisterUnregister End")
        println()
        println("Check EventPriorityTest Start")
        EventPriorityTest()
        println("Check EventPriorityTest End")
        println()
        println("Check EventCancelTest Start")
        EventCancelTest()
        println("Check EventCancelTest End")
        println()
        println("Check ModuleEnableDisableTest Start")
        INSTANCE.module.onKey(1)
        INSTANCE.module.onKey(1)
        println("Check ModuleEnableDisableTest End")
        println()
        println("Check ModuleExtendTest Start")
        INSTANCE.module.onKey(2)
        INSTANCE.module.onKey(2)
        println("Check ModuleExtendTest1 End")
        println()
        INSTANCE.module.onKey(3)
        INSTANCE.module.onKey(3)
        println("Check ModuleExtendTest1 End")
        println()
        println("Check ModuleSettingTest Start")
        INSTANCE.module.onKey(4)
        println("Check ModuleSettingTest End")
        println()
        println("Check ModuleType Start")
        INSTANCE.module.onKey(5)
        println("Check ModuleType End")
        println()
        println("Check CommandTest Start")
        println(INSTANCE.command.execCommand("`t"))
        println(INSTANCE.command.execCommand("`t var1"))
        println(INSTANCE.command.execCommand("`t var1 1"))
        println("Check CommandTest End")
        println()
        println("Check ConfigTest Start")
        ConfigTest()
        println("Check ConfigTest End")
    }
}